service/dap/server.go

```diff
@@ -0,0 +1,464 @@
+	case *dap.SetExceptionBreakpointsRequest:
+		s.onSetExceptionBreakpointsRequest(request)
+	case *dap.ConfigurationDoneRequest:
+		s.onConfigurationDoneRequest(request)
+	case *dap.ContinueRequest:
+		s.onContinueRequest(request)
+	case *dap.NextRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.StepInRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.StepOutRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.StepBackRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.ReverseContinueRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.RestartFrameRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.GotoRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.PauseRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.StackTraceRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.ScopesRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.VariablesRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.SetVariableRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.SetExpressionRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.SourceRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.ThreadsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.TerminateThreadsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.EvaluateRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.StepInTargetsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.GotoTargetsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.CompletionsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.ExceptionInfoRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.LoadedSourcesRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.DataBreakpointInfoRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.SetDataBreakpointsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.ReadMemoryRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.DisassembleRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.CancelRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	case *dap.BreakpointLocationsRequest:
+		s.sendUnsupportedErrorResponse(request.Request)
+	default:
+		// This is a DAP message that go-dap has a struct for, so
+		// decoding succeeded, but this function does not know how
+		// to handle. We should be sending an ErrorResponse, but
+		// we cannot get to Seq and other fields from dap.Message.
+		// TODO(polina): figure out how to handle this better.
+		// Consider adding GetSeq() method to dap.Message interface.
+		panic(fmt.Sprintf("Unable to process %#v", request))
```

Comment:
I'd rather log an error than panic here.

Reply:
I think you should use the logger with loglevel Error (see https://github.com/go-delve/delve/pull/1864)